lib/llm/src/block_manager/block/transfer/nixl.rs

```diff
@@ -136,26 +134,27 @@ where
         None,
     )?;
 
-    let mut xfer_args = OptArgs::new()?;
-
-    if let Some(notify) = notify {
-        xfer_args.set_has_notification(true)?;
-        xfer_args.set_notification_message(notify.as_bytes())?;
+    let still_pending = nixl_agent.post_xfer_req(&xfer_req, None)?;
+
+    if still_pending {
+        Ok(Box::new(Box::pin(async move {
+            let nixl_agent = nixl_agent_arc
+                .as_ref()
+                .as_ref()
+                .expect("NIXL agent not found");
+
+            loop {
+                match nixl_agent.get_xfer_status(&xfer_req) {
+                    Ok(false) => break, // Transfer is complete.
+                    Ok(true) => tokio::time::sleep(std::time::Duration::from_millis(5)).await, // Transfer is still in progress.
```

Comment:
open issue to change the nixl-sys api to avoid using a bool and use a more intuitive enum instead.